presto-hive/src/test/java/io/prestosql/plugin/hive/benchmark/FileFormat.java

```diff
@@ -281,6 +328,86 @@ public abstract FormatWriter createFileFormatWriter(
             HiveCompressionCodec compressionCodec)
             throws IOException;
 
+    public Optional<HivePageSourceFactory> getHivePageSourceFactory(HdfsEnvironment environment)
+    {
+        return Optional.empty();
+    }
+
+    public Optional<HiveRecordCursorProvider> getHiveRecordCursorProvider(HdfsEnvironment environment)
+    {
+        return Optional.empty();
+    }
+
+    public final ConnectorPageSource createFileFormatReader(
```

Comment:
when is `createFileFormatReader` and `createHiveReader` used? They both return `ConnectorPageSource`

Reply:
`createHiveReader ` returns a format-specific reader with the hive connector's wrapper page source. 

`createHiveReader` is used for projection pushdown test, because HivePageSource provides projection adaptation. `createFileFormatReader` is used when we're only considering the format specific pagesource.

Reply:
Maybe `createHiveReader` should be renamed to `createGenericReader`?